netrcutil/netrcutil_test.go

```diff
@@ -76,3 +74,82 @@ func TestCreateFile(t *testing.T) {
 		require.Equal(t, testAppendFileContent, writtenContent)
 	}
 }
+
+func TestCreateOrUpdateFile(t *testing.T) {
+	t.Run("when no .netrc file is present", func(t *testing.T) {
+		netRC, _, cleanup := setup(t)
+		defer cleanup()
+
+		isExists, err := pathutil.IsPathExists(netRC.OutputPth)
+		require.NoError(t, err)
+		require.Equal(t, false, isExists)
+
+		err = netRC.CreateOrUpdateFile(NetRCItemModel{Machine: "testhost.com", Login: "testusername", Password: "testpassword"})
+		require.NoError(t, err)
+
+		isExists, err = pathutil.IsPathExists(netRC.OutputPth)
+		require.NoError(t, err)
+		require.Equal(t, true, isExists)
+
+		writtenContent, err := fileutil.ReadStringFromFile(netRC.OutputPth)
+		require.NoError(t, err)
+		require.Equal(t, testCreateFileContent, writtenContent)
+	})
+
+	t.Run("when a .netrc file is already present", func(t *testing.T) {
+		netRC, tmpDir, cleanup := setup(t)
+		defer cleanup()
+
+		// set up existing .netrc file
+		// a different client is used on the same path
+		// to avoid issues with its internal state like existing .netrc entries
+		netRCForExistingFile := New()
+		netRCForExistingFile.OutputPth = netRC.OutputPth
+		err := netRCForExistingFile.CreateOrUpdateFile(NetRCItemModel{Machine: "testhost.com", Login: "testusername", Password: "testpassword"})
+		require.NoError(t, err)
+
+		isExists, err := pathutil.IsPathExists(netRC.OutputPth)
+		require.NoError(t, err)
+		require.Equal(t, true, isExists)
+
+		err = netRC.CreateOrUpdateFile(NetRCItemModel{Machine: "testhost2.com", Login: "testusername2", Password: "testpassword2"})
+		require.NoError(t, err)
+
+		isExists, err = pathutil.IsPathExists(netRC.OutputPth)
+		require.NoError(t, err)
+		require.Equal(t, true, isExists)
+
+		writtenContent, err := fileutil.ReadStringFromFile(netRC.OutputPth)
+		require.NoError(t, err)
+		require.Equal(t, testAppendFileContent, writtenContent)
+
+		// verify backup file was created
+		backupFile := ""
+		found := 0
+		err = filepath.Walk(tmpDir, func(path string, info os.FileInfo, err error) error {
+			if !info.IsDir() && path != netRC.OutputPth {
+				backupFile = path
+				found += 1
+			}
+			return nil
+		})
+		require.NoError(t, err)
+
+		require.Equal(t, 1, found)
+		backupContent, err := fileutil.ReadStringFromFile(backupFile)
+		require.NoError(t, err)
+		require.Equal(t, testCreateFileContent, backupContent)
+	})
+}
+
+func setup(t *testing.T) (*NetRCModel, string, func()) {
+	tmpDir, err := pathutil.NormalizedOSTempDirPath("__netrc_test__")
```

Comment:
Could we add the `t.Name()` to the path? It would make the corresponding tests parallelizable if we ever want to do that. Not a must have though.

Reply:
This uses `TempDir` under the hood which already adds a random component to the path.

https://pkg.go.dev/io/ioutil#TempDir